### ISSUE TITLE
Resizing Window / Change Resolutions

### ISSUE THREAD

**Issue by Manghi** [ggez/ggez#94]
Is there any way to access the `renderer` in ggez (v0.3.0)? Previously I was on 0.2.x and could get through renderer directly through `Context.renderer`. Something like

```Rust
let ref mut renderer = _ctx.renderer;
let _ = renderer.set_logical_size(w, h);
{
    let window = renderer.window_mut().unwrap();
    let _ = window.set_size(w, h);
}
```

Knowing that things have moved around quite a bit, I cannot find a new way to access it. I have been looking through `sdl_context` as one possibility but I can't seem to figure it out.

**Comment by icefoxen** [comment]
ggez 0.3.0 doesn't use SDL2's renderer at all.  It replaces it with its own drawing backend with gfx-rs.  The SDL2 `Window` is hidden in `ggez::graphics::GraphicsContextGeneric` (see https://github.com/ggez/ggez/blob/master/src/graphics/mod.rs#L177 ) but isn't publicly accessible.

I actually didn't realize SDL2 allowed setting resolution post-setup at all, since I'm used to OpenTK or XNA where this is communicated through a window event of some kind (where you drag the window to the size you want) and SDL2 doesn't support that.  It didn't occur to me to look further and find `window.set_size()`.

This is quite easy to add though, so, I'll add it soon.

**Comment by Manghi** [comment]
I always assumed that window/resolution changes needed to destroy and recreate the window completely, so I too was surprised to see SDL2 allow this on the fly. Currently I can get to window through `gfx_context.get_window()` and successfully wrote something to fiddle with full screen and window size.

I did see that someone had stated 
```Rust
/// Returns a reference to the SDL window.  Ideally you should not need to use this.
pub fn get_window(&mut self) -> &mut sdl2::video::Window {
        &mut self.window
    }
```
Is the intent that ggez should expose those functions without any user digging in there or is there another reason? 

Also I looked through gfx-rs and didn't see anything resembling changing resolution. Might this be a question directed towards their team?

**Comment by icefoxen** [comment]
No, you're asking in the right place.  SDL2 handles the window size, gfx-rs should only handle the process of drawing stuff on it.

The reason for that comment is basically that ggez should supply functions to do anything you would need the SDL2 window to do on its own, so you don't have to dig through the layer of abstraction it provides and touch SDL2 directly.

Resizing a window shouldn't involve destroying and re-creating anything; it's possible to tell an OpenGL context or whatever that its window size has changed.  I think flipping to full screen and back is the only case where it's necessary to shut down and restart everything.

**Comment by Manghi** [comment]
Thanks for the info. If the `gfx-rs` replaced the SDL2 renderer, wouldn't `gfx-rs` need to handle adjustments to the resolution? If you look at my original post there are two basic calls:
1. Update the window size
2. Update the logical size on the renderer (resolution)

I'm still not quite sure how 2 would be supported under `gfx-rs`. Also side note, for what it's worth I was actually able to get fullscreen toggling working on `ggez 0.3.0.` via `&mut _ctx.gfx_context.get_window()`.

**Comment by icefoxen** [comment]
ggez can be told what the logical size of the window is via calling `graphics::set_screen_coordinates()`.  It does appear that gfx-rs handles the window size changing intelligently, though exactly what it does to the logical coordinates (ie, if you make the window smaller, what subsection of it gets zoomed in on) appears somewhat arbitrary.

In some tests, calling SDL2's `window.set_size()` followed by  `graphics::set_screen_coordinates()` appears to have the desired effect, I think.  There may be strange bugs with coordinate systems lurking around the edges.

**Comment by icefoxen** [comment]
Should now be available natively from commit 4609ab28ab893f1b48478369dabde3bc1f6bdcc7 , added a `graphics::set_mode()` function to mirror Love2D's.
